compatibility_lib/compatibility_lib/compatibility_store.py

```diff
@@ -97,6 +97,44 @@ def __eq__(self, o):
                     self.timestamp == o.timestamp)
         return NotImplemented
 
+    def with_updated_dependency_info(
```

Comment:
Would this be used in production code or would this be beneficial for testing only?

Reply:
It could be used for production code but right now I want to use it for testing.

Reply:
If that's the case, I think it makes more sense to house this method in our fake store or as a utility function in the test suite.

Reply:
I think that this method is logical for this class i.e. it makes sense for both test and non-test usage. Right now I plan on using it for tests but it would be equally useful outside of tests.

Reply:
But you just said that you have no plans to use it outside of testing, right? I just find it strange to add this code to production if that's the case.

Reply:
I think that your argument makes sense but I'm not fully convinced ;-)

I think that this method belongs with the class just like `__repr__` (which is usually only used for debug) or any other method.

Reply:
I'm punt this to the rest of the team, since I'm not entirely convinced by that argument either.

Reply:
Given the response on the thread, I'm comfortable with approving these changes!

Reply:
I can change the approach if there is more feedback.